New: cursor and autoFocus

Rivki7 opened the pull request:
Hi, 
I added cursor pointer fot the buttons and auto focus in the forms in the first screen of the app.

baruchiro commented:
Use the "Hide whitespace" to easily review the real change:
![image](https://github.com/brafdlog/caspion/assets/17686879/db23554f-7d20-4ca7-b411-ecaa954a863b)

The rest of the change is because we added a `lint-staged` script which will run on your changed files and format them before the commit. Now when someone touches on a file that wasn't formatted before, he will take the new format with him.

baruchiro commented on ui-react/src/components/topBar/ReportProblemModal.tsx:
```diff
@@ -196,6 +195,7 @@ function ReportProblemModal({ show, onClose }: ReportProblemModalProps) {
                   value={form.title}
                   isInvalid={!!errors.title}
                   onChange={(e) => setField('title', e.target.value)}
+                  autoFocus
```
This is the change

baruchiro commented on ui-react/src/components/accounts/EditImporter.tsx:
```diff
@@ -54,27 +65,54 @@ export default function EditImporter({
+                    onChange={(event) =>
+                      onLoginFieldChanged(loginField, event.target.value)
+                    }
+                    autoFocus={index === 0}
```
This is the change

baruchiro commented on ui-react/src/components/accounts/Account.tsx:
```diff
@@ -8,35 +8,49 @@ import StatusIndicator from './StatusIndicator';
+    (log) => log.originalEvent && log.originalEvent.exportedTransactionsNum > 0,
+  );
   if (!account.active) containerStyles.push(styles.notActive);
+  if (!actionButtons) containerStyles.push(styles.pointer);
```
This is the change.

Although I'm not sute why you need this if, can you explain?

baruchiro commented:
Do you think you can cursor these too?
<img width="208" alt="image" src="https://github.com/brafdlog/caspion/assets/17686879/cb068c65-616a-4ced-a8cf-260307c1d289">

Rivki7 commented:
> Use the "Hide whitespace" to easily review the real change: ![image](https://private-user-images.githubusercontent.com/17686879/308485999-db23554f-7d20-4ca7-b411-ecaa954a863b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkyMDIwMTUsIm5iZiI6MTcwOTIwMTcxNSwicGF0aCI6Ii8xNzY4Njg3OS8zMDg0ODU5OTktZGIyMzU1NGYtN2QyMC00Y2E3LWI0MTEtZWNhYTk1NGE4NjNiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI5VDEwMTUxNVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTZlMWE1ZDEwNTRmYzdjN2ZjNjg2NzJmZjhhZmJiMTcxMTY3Mjk1OTZjMjcxMTAxYThiZDYxYmQ0OWE5NWU4ODQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bjeCF8SYzuXg3puQEZBLZopJ4IcnOCVyKiwDTKWDdgs)
> 
> The rest of the change is because we added a `lint-staged` script which will run on your changed files and format them before the commit. Now when someone touches on a file that wasn't formatted before, he will take the new format with him.

Good to know that, thanks!

Rivki7 commented:
> Do you think you can cursor these too? <img alt="image" width="208" src="https://private-user-images.githubusercontent.com/17686879/308523805-cb068c65-616a-4ced-a8cf-260307c1d289.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MDkyMDcxMDcsIm5iZiI6MTcwOTIwNjgwNywicGF0aCI6Ii8xNzY4Njg3OS8zMDg1MjM4MDUtY2IwNjhjNjUtNjE2YS00Y2VkLWE4Y2YtMjYwMzA3YzFkMjg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAyMjklMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMjI5VDExNDAwN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTUzNDg5YjI5NDQxNDc5YTA3NjRmMDViZWU5MjliMmVlNGZjZTkxYWY1NGI0OTBkNGMwMzFiNTMxMTk4ODc1ZDUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0._D8e9bB8C8Xf3W6T1UtIPY8UJtCtZsbk2WEZ9T45C9I">

I added that to the settings icon now (and also added autoFocus to the setting modal first text area). 
I tried to add that to the switch input as well but it is too complicated because of bootstrap. I consulted with Gemini and gpt and couldn't find a simple way to do that without changing the form structure.

baruchiro commented on ui-react/src/components/Body.tsx:
```diff
@@ -65,58 +69,112 @@ const Body = ({ scrape }: BodyProps) => {
+        >
+          הפעל
+        </Button>
+        <Image
+          src={settingsIcon}
+          onClick={() => showModal(null, ModalStatus.GENERAL_SETTINGS)}
+          className={styles.pointer}
+        />
         <Form.Check
```
This is the change

baruchiro reviewed (approved):
Thank you!

![ThankYouThanksGIF](https://github.com/brafdlog/caspion/assets/17686879/78f81b28-be92-4b6e-8094-01f85bd34a0e)
